### PR TITLE
Tie post endpoint with frontend

### PR DESCRIPTION
**Pull request by IlyasAli1**
https://trello.com/c/VXdDADar

### CODE REVIEW

File: WhaleSpotting.UnitTests/Controllers/SightingControllerTests.cs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ﻿using FakeItEasy;
 using FluentAssertions;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
```
Comment (oskarwilliams) [inline WhaleSpotting.UnitTests/Controllers/SightingControllerTests.cs]: Do we need this?

File: WhaleSpotting.UnitTests/Controllers/SightingControllerTests.cs [hunk 2]
```diff
@@ -117,8 +118,10 @@ public void CreateSighting_CalledWithInvalidNewSighting_ReturnsBadRequest()
             var response = _underTest.CreateSighting(newSighting);
 
             // Assert
-            var badRequestResult = response.Should().BeOfType<BadRequestObjectResult>().Subject;
-            badRequestResult.Value.Should().Be(exceptionMessage);
+            var validationErrorResult = response.Should().BeOfType<ObjectResult>();
+            var validationProblemDetails = validationErrorResult.Subject.Value as ValidationProblemDetails;
```
Comment (oskarwilliams) [inline WhaleSpotting.UnitTests/Controllers/SightingControllerTests.cs]:
```suggestion
            var validationErrorResult = response.Should().BeOfType<ObjectResult>().Subject;
            var validationProblemDetails = validationErrorResult.Value.Should().BeOfType<ValidationProblemDetails>().Subject;
```

File: WhaleSpotting/Controllers/SightingsController.cs [hunk 3]
```diff
@@ -11,6 +11,7 @@ namespace WhaleSpotting.Controllers
 {
     [ApiController]
     [Route("[controller]")]
+    //[ValidateModel]
```
Comment (oskarwilliams) [inline WhaleSpotting/Controllers/SightingsController.cs]: Commented code

File: WhaleSpotting/Controllers/SightingsController.cs [hunk 4]
```diff
@@ -33,11 +34,11 @@ public IActionResult CreateSighting([FromBody] SightingRequestModel sightingRequ
             {
                 var newSighting = _sightings.CreateSighting(sightingRequestModel);
                 return Created($"/sighting/{newSighting.Id}", newSighting);
-                // TODO note the url parameter above is to be updated if a get sighting by id endpoint is created. 
             }
             catch (Exception e)
             {
-                return BadRequest(e.Message);
+                ModelState.AddModelError("Thrown Error", e.Message);
```
Comment (oskarwilliams) [inline WhaleSpotting/Controllers/SightingsController.cs]:
```suggestion
                ModelState.AddModelError(nameof(SightingRequestModel.SightedAt), e.Message);
```
We know what this error is on, so we should say that in the key

File: WhaleSpotting/Controllers/SightingsController.cs [hunk 5]
```diff
@@ -49,4 +50,6 @@ public async Task<ActionResult<SightingResponseModel>> ConfirmSighting([FromRout
             return sighting == null ? NotFound() : sighting;
         }
     }
+
+
```
Comment (oskarwilliams) [inline WhaleSpotting/Controllers/SightingsController.cs]: New lines

File: WhaleSpotting/react-app/src/components/ReportSighting.tsx [hunk 6]
```diff
@@ -1,91 +1,161 @@
...
-    const [date, setDate] = useState("");
-    const [location, setLocation] = useState("");
-    const [species, setSpecies] = useState("");
-    const [quantity, setQuantity] = useState("");
-    const [longitude, setLongitude] = useState("");
-    const [latitude, setLatitude] = useState("");
-    const [orcaPod, setOrcaPod] = useState("");
-    const [imageUrl, setImageUrl] = useState("");
-    const [description, setDescription] = useState("");
+export function ShowResultMessage({ responseMessage } : any): JSX.Element {
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/components/ReportSighting.tsx]: This should be in another component :) Then both can be export default. Also we should have an interface for the props

File: WhaleSpotting/react-app/src/components/ReportSighting.tsx [hunk 7]
```diff
@@ -1,91 +1,161 @@
...
+                    isError = true;
+                    return r.json();
+                }
+                return r.json();
+            })
+            .catch(r => {
+                isError = true;
+                return r.json();
+            });
+        if (isError) {
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/components/ReportSighting.tsx:54]: new line before this please :)

File: WhaleSpotting/react-app/src/apiModels/CreateSightingApiModel.ts [hunk 8]
```diff
@@ -28,11 +28,10 @@ export interface CreateSightingApiModel {
     species: Species;
     quantity: number;
     description: string;
-    longitude: number;
-    latitude: number;
+    longitude: number | null;
+    latitude: number | null;
     location: string;
-    sightedAt: Date;
+    sightedAt: Date | undefined;
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/apiModels/CreateSightingApiModel.ts]: Should this be allowed to be undefined?

**Review by oskarwilliams — CHANGES_REQUESTED** [review]

File: WhaleSpotting.UnitTests/Controllers/SightingControllerTests.cs (hunk 1 above)
Comment (OptimumCOW) [reply WhaleSpotting.UnitTests/Controllers/SightingControllerTests.cs]: Addressed.

File: WhaleSpotting.UnitTests/Controllers/SightingControllerTests.cs (hunk 2 above)
Comment (OptimumCOW) [reply WhaleSpotting.UnitTests/Controllers/SightingControllerTests.cs]: Addressed.

File: WhaleSpotting/Controllers/SightingsController.cs (hunk 3 above)
Comment (OptimumCOW) [reply WhaleSpotting/Controllers/SightingsController.cs]: Addressed.

File: WhaleSpotting/Controllers/SightingsController.cs (hunk 4 above)
Comment (OptimumCOW) [reply WhaleSpotting/Controllers/SightingsController.cs]: Addressed.

File: WhaleSpotting/Controllers/SightingsController.cs (hunk 5 above)
Comment (OptimumCOW) [reply WhaleSpotting/Controllers/SightingsController.cs]: Addressed.

File: WhaleSpotting/react-app/src/components/ReportSighting.tsx (hunk 7 above)
Comment (OptimumCOW) [reply WhaleSpotting/react-app/src/components/ReportSighting.tsx:54]: Addressed.

File: WhaleSpotting/react-app/src/apiModels/CreateSightingApiModel.ts (hunk 8 above)
Comment (OptimumCOW) [reply WhaleSpotting/react-app/src/apiModels/CreateSightingApiModel.ts]: Addressed.

File: WhaleSpotting/react-app/src/components/ShowResultMessage.tsx [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+export default function ShowResultMessage({ responseMessage }: ShowResultMessageProps): JSX.Element {
+    if (responseMessage) {
+        return (
+            <p data-testid="response-result" className="response-message-card">
+                {responseMessage}
+            </p>);
+    }
+    return <div></div>;
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/components/ShowResultMessage.tsx]:
```suggestion
    return responseMessage 
        ? <p data-testid="response-result" className="response-message-card">
            {responseMessage}
        </p>
        : <div/>;
}
```

File: WhaleSpotting/react-app/src/styles/Buttons.scss [hunk 10]
```diff
@@ -5,6 +5,7 @@
     color: $textOnPrimary;
     border-radius: 10px;
     border: 0;
+    height:50px;
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/styles/Buttons.scss]: space before value pls

File: WhaleSpotting/react-app/src/styles/Buttons.scss [hunk 11]
```diff
@@ -33,6 +34,7 @@
     text-decoration: none;
     text-align: center;
     padding: 15px 0px;
+    height:unset;
```
Comment (oskarwilliams) [inline WhaleSpotting/react-app/src/styles/Buttons.scss]: here too

**Review by oskarwilliams — CHANGES_REQUESTED** [review]

File: WhaleSpotting/react-app/src/components/ReportSighting.tsx (hunk 6 above)
Comment (OptimumCOW) [reply WhaleSpotting/react-app/src/components/ReportSighting.tsx]: Addressed.

File: WhaleSpotting/react-app/src/components/ShowResultMessage.tsx (hunk 9 above)
Comment (IlyasAli1) [reply WhaleSpotting/react-app/src/components/ShowResultMessage.tsx]: Done

File: WhaleSpotting/react-app/src/styles/Buttons.scss (hunk 10 above)
Comment (IlyasAli1) [reply WhaleSpotting/react-app/src/styles/Buttons.scss]: Done

File: WhaleSpotting/react-app/src/styles/Buttons.scss (hunk 11 above)
Comment (IlyasAli1) [reply WhaleSpotting/react-app/src/styles/Buttons.scss]: Done

**Review by oskarwilliams — APPROVED** [review]

**Pull request merged**